live/startup

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env sh
+
+FIREFOX_PROFILE=$HOME/.mozilla/firefox/profile
+COOKIES_DB=$FIREFOX_PROFILE/cookies.sqlite
+TOKEN_FILE=/run/agama/token
+
+if [ ! -f $COOKIES_DB ]; then
+  mkdir -p $FIREFOX_PROFILE
+  sqlite3 $COOKIES_DB "CREATE TABLE moz_cookies (
+    id INTEGER PRIMARY KEY,
+    originAttributes TEXT NOT NULL DEFAULT '',
+    name TEXT,
+    value TEXT,
+    host TEXT,
+    path TEXT,
+    expiry INTEGER,
+    lastAccessed INTEGER,
+    creationTime INTEGER,
+    isSecure INTEGER,
+    isHttpOnly INTEGER,
+    inBrowserElement INTEGER DEFAULT 0,
+    sameSite INTEGER DEFAULT 0,
+    rawSameSite INTEGER DEFAULT 0,
+    schemeMap INTEGER DEFAULT 0,
+    CONSTRAINT moz_uniqueid UNIQUE (name, host, path, originAttributes));"
+else
+  sqlite3 $COOKIES_DB "DELETE FROM moz_cookies WHERE name = 'agamaToken';"
+fi
+
+NOW=$(date +%s)
+VALUE=$(cat $TOKEN_FILE)
+EXPIRATION=$(date --date '1 day' +%s)
+
+sqlite3 $COOKIES_DB "INSERT INTO moz_cookies \
```

Comment:
I do not like hacking the internal Firefox cookies database directly. That's very fragile and works only with Firefox.

What about creating a `/run/agama/token.html` file with root-only permissions with content like 

```html
<!DOCTYPE html>
<html>
<body>
  <script>
    document.location = "http://localhost/login?token=" + <JS-escaped-token-value>;
  </script>
</body>
</html>
```

and implement `/login` endpoint which would set the `agamaToken` cookie and redirect to `/`.

And then use that file in the `firefox` command below like

```sh
firefox --kiosk --profile $HOME/.mozilla/firefox/profile /run/agama/token.html
```

That special file would avoid exposing the token in the process list (in contrast when using the token directly in the firefox command).

The advantage would not hacking the SQLite internals and would be generic for all browsers.

Note: Originally I wanted to put something like 
```js
 document.cookie = "agamaToken=" + tokenValue + ";domain=localhost;path=/api";
```
into that HTML file. But for security reasons it is not possible to set cookie for a different domain...

Reply:
Of course, I am open to other approaches and I see your point. However bear in mind that we only need to support Firefox and, if something changes, we can detect soon enough in our CI.

To be honest, I would rather prefer some kind of API to import the cookie and do not have to extend the backend (and add a redirection). But I am not against this solution.

Reply:
Thinking about your proposal, given that we need to extend the backend anyway, we can get rid of the `token.html` file by just setting the cookie *and* redirecting the user. The following code seems to work:

```rust
pub async fn get_cookie(
    State(state): State<ServiceState>,
    Query(params): Query<GetCookieParams>,
) -> impl IntoResponse {
    let decoding = DecodingKey::from_secret(state.config.jwt_secret.as_ref());
    let mut headers = HeaderMap::new();

    if jsonwebtoken::decode::<TokenClaims>(&params.token, &decoding, &Validation::default()).is_ok()
    {
        let cookie = format!("agamaToken={}; HttpOnly", params.token);
        headers.insert(
            SET_COOKIE,
            cookie.parse().expect("could not build a valid cookie"),
        );
    }

    headers.insert(LOCATION, "/".parse().unwrap());
    (StatusCode::PERMANENT_REDIRECT, headers)
}
```

If it is OK for you, I will clean-up the code a bit and update the PR.